bee-network/src/swarm/protocols/gossip/upgrade.rs

```diff
@@ -0,0 +1,66 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::init::global::network_id;
+
+use futures::{future, AsyncRead, AsyncWrite};
+use libp2p::{core::UpgradeInfo, InboundUpgrade, OutboundUpgrade};
+use log::trace;
+
+use std::iter;
+
+const PROTOCOL_INFO_IOTA_GOSSIP: &str = "iota-gossip";
+const PROTOCOL_INFO_VERSION: &str = "1.0.0";
+
+/// Configuration for an upgrade to the `IotaGossip` protocol.
+#[derive(Debug, Clone, Default)]
+pub struct GossipUpgrade;
+
+impl UpgradeInfo for GossipUpgrade {
+    type Info = Vec<u8>;
+    type InfoIter = iter::Once<Self::Info>;
+
+    fn protocol_info(&self) -> Self::InfoIter {
+        iter::once(
+            format!(
+                "/{}/{}/{}",
+                PROTOCOL_INFO_IOTA_GOSSIP,
+                network_id(),
```

Comment:
Why does this need to be a global ? Can't we have it as a field of `GossipUpgrade` ?

Reply:
If you do that then you have to also add it to `GossipHandler`, because it's `libp2p` which creates an instance of that type. So it increases the complexity of two types just for the sake of not using a constant global for now benefit at all (because we use global constants everywhere in our codebase. It's just that here, this constant can only be initialized at runtime). I made sure through the `OnceCell` type, that this can never ever become a problem.